Podcast/SearchViewController.swift

```diff
@@ -240,21 +241,36 @@ class SearchViewController: ViewController, UISearchControllerDelegate, UITableV
             }
             return
         }
-        if searchText == lastSearchText { return }
-        lastSearchText = searchText
-        tabUnderlineView.isHidden = false
-        searchResultsTableView.isHidden = false
-        pastSearchesTableView.isHidden = true
-        didDismissItunesHeaderForQuery = false
-        addSearchITunesHeader()
-        mainScrollView = searchResultsTableView
-        searchResultsTableView.startLoadingAnimation()
-        for data in tableViewData {
-            data.fetchData(query: searchText, newSearch: true)
+        if self.lastSearchText == searchText && pastSearchesTableView.isHidden { return }
+        self.lastSearchText = searchText
+        self.tabUnderlineView.isHidden = false
+        self.searchResultsTableView.isHidden = false
+        self.pastSearchesTableView.isHidden = true
+        self.didDismissItunesHeaderForQuery = false
+        self.addSearchITunesHeader()
+        self.mainScrollView = self.searchResultsTableView
+        self.searchResultsTableView.startLoadingAnimation()
+        for data in self.tableViewData {
+            data.searchResults = []
+        }
+
+        // put a timer on searching so not overloading with requests
+        if let timer = searchDelayTimer {
+            timer.invalidate()
+            searchDelayTimer = nil
+        }
+        searchDelayTimer = Timer.scheduledTimer(timeInterval: 0.25, target: self, selector: #selector(searchAfterDelay), userInfo: nil, repeats: false)
+
+        self.searchResultsTableView.reloadData()
+        self.searchResultsTableView.loadingAnimation.bringSubview(toFront: self.searchResultsTableView)
+        self.updateTableViewInsetsForAccessoryView()
+
+    }
+
+    @objc func searchAfterDelay() {
+        for data in self.tableViewData {
+            data.fetchData(query: lastSearchText, newSearch: true)
```

Comment:
the "no results found" screen shows up for a little bit after the loading indicator spins for a bit, and then shows results (only for series and episodes):
![image](https://user-images.githubusercontent.com/20246620/35841037-c68ddca4-0ac7-11e8-8e7d-cc42b6786915.png)